### ISSUE TITLE
Get/save predictions made during evaluation for object detection models

### ISSUE THREAD

**Issue by Shubhambindal2017** [tensorflow/models#10264]
At present, if someone wants to evaluate and get/save predictions on same dataset for a trained model, they need to run evaluation (by running model_main_tf2.py with checkpoint_dir arguement) and inference separately, so what if the user gets an option to save the prediction during the evaluation itself, it can reduce need to run inference separately.

And a queries:
I have fine-tuned an SSD-Mobilenetv2 with train config fixed resize 300x300 built using tensorflow objection detection API and saved in TF Saved_Model format.
Questions:
 - How during inference it is able to accept input images of any shape (not 300x300) without the need for any preprocessing to resize to 300x300 first and then pass to the model. 
 - Does it is because saved_model by default does resize during inference? (If yes, does it also normalize them because before doing convolution operations) (I am new to saved_model format but I think it is not because of saved_model, but then how is it possible - as I think SSD-Mobilenet includes FC layers which require fixed input size) OR does the architecture uses AdaptivePooling in b/w to achieve this.
